org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/makestatic/ContextCalculator.java

```diff
@@ -0,0 +1,253 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse
+ * Public License 2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors: Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.internal.corext.refactoring.code.makestatic;
+
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ASTParser;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.ExpressionStatement;
+import org.eclipse.jdt.core.dom.IMethodBinding;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.MethodInvocation;
+import org.eclipse.jdt.core.dom.NodeFinder;
+
+import org.eclipse.jdt.internal.corext.dom.IASTSharedValues;
+import org.eclipse.jdt.internal.corext.dom.Selection;
+import org.eclipse.jdt.internal.corext.refactoring.structure.ASTNodeSearchUtil;
+
+/**
+ * This class calculates and provides context for a static method refactoring process. It processes
+ * the given input (SelectionFields), which can be an ICompilationUnit or an IMethod, and calculates
+ * the corresponding TargetFields, which are used in the actual refactoring process.
+ */
+public class ContextCalculator {
+
+	private SelectionInputType fSelectionInputType;
+
+	private Selection fSelectionEditorText;
+
+	private ICompilationUnit fSelectionICompilationUnit;
+
+	private CompilationUnit fSelectionCompilationUnit;
+
+	private ASTNode fSelectionASTNode;
+
+	/**
+	 * The {@code IMethodBinding} object representing the binding of the refactored method.
+	 */
+	private IMethodBinding fTargetIMethodBinding;
+
+	/**
+	 * The {@code IMethod} object representing the selected method on which the refactoring should
+	 * be performed.
+	 */
+	private IMethod fTargetIMethod;
```

Comment:
Normally we did not document private fields but in the process of creating the ContextCalculator and deleting some redundant fields of MakeStaticRefactoring I transfered the already existing javaDocs of some fields from MakeStaticRefactoring to ContextCalculator. I will also document the other fields because it could benefit the next person that wants to debug our code.

Reply:
Done. 85eeeac64e